src/main/java/com/minecolonies/colony/WorkManager.java

```diff
@@ -211,13 +233,38 @@ public void onWorldTick(TickEvent.WorldTickEvent event)
                 if (!o.isValid(colony))
                 {
                     iter.remove();
+                    dirty = true;
+                }
+                else if(o.hasChanged())
+                {
+                    dirty = true;
+                    o.resetChange();
                 }
             }
 
             if ((event.world.getWorldTime() % WORK_ORDER_FULFILL_INCREMENT) == 0)
             {
-                workOrders.values().stream().filter(o -> !o.isClaimed()).forEach(o -> o.attemptToFulfill(colony));
+                workOrders.values().stream().filter(o -> !o.isClaimed()).sorted((first, second) -> second.getPriority() > first.getPriority() ? 1 : (second.getPriority() < first.getPriority() ? -1 : 0))
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 202 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00103)